packages/tools/viewer/src/viewerElement.ts

```diff
@@ -1347,62 +1398,25 @@ export abstract class ViewerElement<ViewerClass extends Viewer = Viewer> extends
     }
 
     /**
-     * Calculates the alpha, beta, and radius along with the target point to create a HotSpot from a camera.
-     * The target point is determined based on the camera's forward ray:
-     *   - If an intersection with the main model is found, the first hit point is used as the target.
-     *   - If no intersection is detected, a fallback target is calculated by projecting
-     *     the distance between the camera and the main model's center along the forward ray.
-     *
-     * @param camera The reference camera used to computes alpha, beta, radius and target
-     * @returns A HotSpot, or null if no model found
+     * Creates a world HotSpot from a camera.
+     * @param camera The camera to create a HotSpot from.
+     * @returns A HotSpot created from the camera.
      */
-    private async _cameraToHotSpot(camera: Camera): Promise<Nullable<HotSpot>> {
+    protected async _createHotSpotFromCamera(camera: Camera): Promise<Nullable<HotSpot>> {
         if (camera instanceof ArcRotateCamera) {
             const targetArray = camera.target.asArray();
             return { type: "world", position: targetArray, normal: targetArray, cameraOrbit: [camera.alpha, camera.beta, camera.radius] };
         }
 
-        await import("core/Culling/ray");
-        const ray = camera.getForwardRay(100, camera.getWorldMatrix(), camera.globalPosition); // Set starting point to camera global position
-        const camGlobalPos = camera.globalPosition.clone();
-
-        if (this.viewerDetails?.scene) {
-            const scene = this.viewerDetails.scene;
-            const model = this.viewerDetails.model;
+        if (this._viewerDetails?.scene) {
+            const scene = this._viewerDetails.scene;
+            const model = this._viewerDetails.model;
 
-            // Target
-            let radius: number = 0.0001; // Just to avoid division by zero
-            const targetPoint = this._tempVectors[0];
+            const selectedAnimation = this.selectedAnimation ?? 0;
+            const worldBounds = model?.getWorldBounds(selectedAnimation);
+            const centerArray = worldBounds ? worldBounds.center : [0, 0, 0];

```

Comment:
Since `CreateHotSpotFromCamera` is already Viewer specific, could we pass in a `Model` instead of a `Scene` + `MeshPredicate`? If we did that and also:
1. Expose `selectedAnimation` from `Model` (rather than `_selectedAnimation` from `ModelInternal`)
2. Change `Model.getWorldBounds` to default to `selectedAnimation` (or just always use `selectedAnimation`)
Then it seems like all this logic could just be in the helper function, which would make it even easier to use.